src/safe-unsafe-meaning.md

```diff
@@ -122,24 +124,35 @@ The decision of whether to mark a trait `unsafe` is an API design choice.
 Rust has traditionally avoided doing this because it makes Unsafe
 Rust pervasive, which isn't desirable. `Send` and `Sync` are marked unsafe
 because thread safety is a *fundamental property* that unsafe code can't
-possibly hope to defend against in the way it could defend against a bad
-`Ord` implementation. The decision of whether to mark your own traits `unsafe`
-depends on the same sort of consideration. If `unsafe` code can't reasonably
-expect to defend against a bad implementation of the trait, then marking the
-trait `unsafe` is a reasonable choice.
+possibly hope to defend against in the way it could defend against a buggy
+`Ord` implementation. Similarly, `GlobalAllocator` is keeping accounts of all
+the memory in the program and other things like `Box` or `Vec` build on top of
+it. If it does something weird (giving the same chunk of memory to another
+request when it is still in use), there's no chance to detect that and do
+anything about it.
+
+The decision of whether to mark your own traits `unsafe` depends on the same
+sort of consideration. If `unsafe` code can't reasonably expect to defend
+against a broken implementation of the trait, then marking the trait `unsafe` is
+a reasonable choice.
 
 As an aside, while `Send` and `Sync` are `unsafe` traits, they are *also*
 automatically implemented for types when such derivations are provably safe
 to do. `Send` is automatically derived for all types composed only of values
 whose types also implement `Send`. `Sync` is automatically derived for all
 types composed only of values whose types also implement `Sync`. This minimizes
-the pervasive unsafety of making these two traits `unsafe`.
+the pervasive unsafety of making these two traits `unsafe`. And not many people
+are going to *implement* memory allocators (not even use directly, for that
+matter).
 
 This is the balance between Safe and Unsafe Rust. The separation is designed to
 make using Safe Rust as ergonomic as possible, but requires extra effort and
 care when writing Unsafe Rust. The rest of this book is largely a discussion
 of the sort of care that must be taken, and what contracts Unsafe Rust must uphold.
 
+[`Send`]: https://doc.rust-lang.org/std/marker/trait.Send.html
```

Comment:
Makes this point to the same version of the stdlib docs that the visible copy of the nomicon does (stable -> stable, nightly -> nightly, etc)

```suggestion
[`Send`]: ../std/marker/trait.Send.html
```